### PR TITLE
fix(scanner): continue walk after special files (fixes #9872)

### LINKED ISSUE #9872: When socket files are in a folder that needs to be synced on linux, some files were not synced until the socket files were removed
### What happened?

OS Fedora 41 
folder to sync is a subfolder in user home containing some socket files of gpg agent (socket files mistakenly got there).

some other  files which were recently changed were randomly not synced until the socket files were removed. 

rescanning and restarting syncthing didn't help.

after removing the socket files the sync process sync all remaining files on 1st attempt.

my assumption is: when the socket files were accessed by the (not so randomly choosing) syncing process the syncing process crashed silently. 

the missing files would have been accessed by the syncing process in the order after the  socket files. since the process crashed silently the files could not get synced. 

- just an assumption -

### Syncthing version

v1.28.0

### Platform & operating system

Linux amd64

### Browser version

Firefox

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by calmh**
We must skip unix sockets, fifos, etc when scanning as these are not filetypes we can handle. Currently we return a "bug" error, which results in the walk being aborted and the rest of the tree being essentially invisible to Syncthing. Instead, just ignore these files and continue onwards.

This might well be #9859 as well but I can't confirm.

### CODE REVIEW

**Comment by acolomb** [comment]
Isn't it a bug in itself that when this function returns the error, it doesn't show up anywhere?  AFAICS the `handleError()` function should be called if `handleItem()` fails, just as it is for errors from `Lstat()` and other calls.

**Comment by calmh** [comment]
That is in fact also a bug, I think, yes

**Comment by calmh** [comment]
The error handling is apparently a bit more subtle than that, so leaving it for the future.

**Review by acolomb — APPROVED** [review]

**Comment by acolomb** [comment]
It is a bit hard to follow how error handling works between all those walk functions and between the `scanner` and `fs` packages.  Hope you'll have more luck wrapping your head around it.  I do think not having any clue in the log for these cases is pretty serious and should be fixed soon.

**Comment by imsodin** [comment]
The problem is that we bubble up unexpected errors that abort scanning/walking, but ignore them at the top. The `handleError` mecchanism is for file-level errors that do not abort walking/scanning.

**Comment by calmh** [comment]
I also suspect that we have errors here that do not merit stopping the walk...

**Comment by imsodin** [comment]
I started to do something about both aspects, but as always with error handling and human readable strings there are so many choices and tradeoffs that combined with my desire to do it "syncthing grade quality :)" stopped me from posting the PR so far - I hope to get there in some Christmas downtime :)
